python_src/src/lamp_py/postgres/postgres_utils.py

```diff
@@ -266,6 +277,44 @@ def select_as_list(
         with self.session.begin() as cursor:
             return [row._asdict() for row in cursor.execute(select_query)]
 
+    def write_to_parquet(
+        self,
+        select_query: sa.sql.selectable.Select,
+        write_path: str,
+        schema: pyarrow.schema,
+        batch_size: int = 1024 * 1024,
+    ) -> str:
+        """
+        stream db query results to parquet file in batches
+
+        this function is meant to limit memory usage when creating very large
+        parquet files from db SELECT
+
+        default batch_size of 1024*1024 is based on "row_group_size" parameter
+        of ParquetWriter.write_batch(): row group size will be the minimum of
+        the RecordBatch size and 1024 * 1024. If set larger
+        than 64Mi then 64Mi will be used instead.
+        https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html#pyarrow.parquet.ParquetWriter.write_batch
+
+        :param select_query: query to execute
+        :param write_path: local file path for resulting parquet file
+        :param schema: schema of parquet file from select query
+        :param batch_size: number of records to stream from db per batch
+
+        :return local path to created parquet file
+        """
+        with self.session.begin() as cursor:
+            result = cursor.execute(select_query).yield_per(batch_size)
+            with pq.ParquetWriter(write_path, schema=schema) as pq_writer:
+                for part in result.partitions():
+                    pq_writer.write_batch(
+                        pyarrow.RecordBatch.from_pylist(
+                            [row._asdict() for row in part], schema=schema
+                        )
+                    )
+
+        return write_path
```

Comment:
i don't think we need to return the write path since it was provided as part of the input.